tesp/luigi_db_utils.py

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env python
+
+"""
+Utilities for accessing and querying the task history database that
+the luigi scheduler can output.
+Requires the luigi config to have the *task_history* defined.
+Eg:
+[task_history]
+db_connection = sqlite:///luigi-task-hist.db
+"""
+
+from pathlib import Path
+import sqlite3
+import pandas
+import click
+
+
+def read_task_db(fname):
+    """
+    Read a task history database written by luigi.
+    """
+    connection = sqlite3.connect(fname)
+    # cursor = connection.cursor()
+    # query = "SELECT name FROM sqlite_master WHERE type='table';"
+    # tables = cursor.execute(query).fetchall()
+
+    tasks = pandas.read_sql_query("SELECT * from tasks", connection)
+    events = pandas.read_sql_query("SELECT * from task_events", connection)
+    params = pandas.read_sql_query("SELECT * from task_parameters", connection)
+
+    return tasks, events, params
+
+
+def retrieve_status(fname, task_name=None):
+    """
+    Retrieve the task status given by `task_name` for each L1 dataset.
+    """
+    if task_name is None:
+        raise ValueError('Parameter "task_name" cannot be None')
```

Comment:
Why supply a default value then? 😄

Reply:
good point. no idea

Reply:
will remove the default of `None`.  I probably had an idea of using a default task name, and for some reason didn't go ahead with that.

Reply:
Cleaned up.